cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/BalancingAction.java

```diff
@@ -161,6 +217,10 @@ public ActionType balancingAction() {
     proposalMap.put(DESTINATION_BROKER_ID, _destinationBrokerId);
     proposalMap.put(DESTINATION_TOPIC_PARTITION, _destinationTp);
     proposalMap.put(ACTION_TYPE, _actionType);
+    if (_sourceBrokerLogdir != null) {
```

Comment:
maybe add a comment here or above stating `_sourceBrokerLogdir` and `_destinationBrokerLogdir` must both be null or neither null, thus checking ``_sourceBrokerLogdir != null` is sufficient.